src/aat/resources/features/F-045/S-151.td.json

```diff
@@ -0,0 +1,47 @@
+{
+  "_guid_": "S-151",
+  "title": "must return 201 if the grant is successful for a user to a valid case ID",
+  "specs": [
+    "a detailed profile in CCD",
+    "has a valid case-reference in input parameters",
+    "has the 201 return code"
+  ],
+  "productName": "CCD Data Store",
+  "operationName": "Grant access to case",
+  "method": "POST",
+  "uri": "/caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/{cid}/users",
+  "user": {
+    "username": "[[$CCD_CASEWORKER_AUTOTEST_EMAIL]]",
+    "password": "[[$CCD_CASEWORKER_AUTOTEST_PASSWORD]]"
+  },
+  "request": {
+    "headers": {
+      "Authorization": "[[DYNAMIC]]",
+      "ServiceAuthorization": "[[DYNAMIC]]",
+      "Content-Type": "application/json"
+    },
+    "pathVariables": {
+      "uid": "[[DYNAMIC]]",
+      "jid": "AUTOTEST1",
+      "ctid": "AAT",
+      "cid": "1573751648840703"
+    },
+    "body": {
+      "id": "Test User"
+    }
+  },
+    "expectedResponse": {
+      "responseCode": 201,
+      "responseMessage": "Created",
+      "headers" : {
+        "Cache-Control" : "no-cache, no-store, max-age=0, must-revalidate",
+        "Content-Length" : "[[ANY]]",
+        "Date" : "[[ANY]]",
+        "Expires" : "0",
+        "Pragma" : "no-cache",
+        "X-Content-Type-Options" : "nosniff",
+        "X-Frame-Options" : "DENY",
+        "X-XSS-Protection" : "1; mode=block"
+      }
+    }
+}
```

Comment:
It looks like a lot of these lines are repeated in all tests. So, you can apply inheritance mechanism, by defining some common objects and extending them.

Reply:
Actually, I had to place the content of Base root file and Base_Expecting_201 in this input file because there are some response headers missing in this 201 response. I am not sure whether all other 201 are going to behave similarly hence I removed the base file extension for this scenario.

Reply:
If I inherit 201 base file then it is breaking the test because headers defined in the base file are not optional and we cannot ignore them, so I had to remove the inheritance.

Reply:
You can still apply a lot of inheritance. Let's do it in a workshop call.

Reply:
All json files are reformatted for leveraging inheritance from common base files.